artofillusion/polymesh/UVMappingCanvas.java

```diff
@@ -272,69 +297,198 @@ private void doPaint(Graphics2D g, boolean export) {
             }
         }
         g.setStroke(normal);
-        if (!export) {
-            for (int i = 0; i < verticesPoints.length; i++) {
-                if (selected[i]) {
-                    if (mappingData.meshes[currentPiece].vertices[mappingData.verticesTable[currentPiece][i]].pinned)
-                        g.setColor(pinnedSelectedColor);
-                    else
-                        g.setColor(selectedColor);
-
-                    g.drawOval(verticesPoints[i].x - 3,
-                            verticesPoints[i].y - 3, 6, 6);
-                } else {
-                    if (mappingData.meshes[currentPiece].vertices[mappingData.verticesTable[currentPiece][i]].pinned)
-                        g.setColor(pinnedColor);
-                    else
-                        g.setColor(unselectedColor);
-
-                    g.fillOval(verticesPoints[i].x - 3,
-                            verticesPoints[i].y - 3, 6, 6);
+        for (int i = 0; i < verticesPoints.length; i++) {
+            if (selected[i]) {
+                if (mappingData.meshes[currentPiece].vertices[mappingData.verticesTable[currentPiece][i]].pinned)
+                    g.setColor(pinnedSelectedColor);
+                else
+                    g.setColor(selectedColor);
+
+                g.drawOval(verticesPoints[i].x - 3,
+                        verticesPoints[i].y - 3, 6, 6);
+            } else {
+                if (mappingData.meshes[currentPiece].vertices[mappingData.verticesTable[currentPiece][i]].pinned)
+                    g.setColor(pinnedColor);
+                else
+                    g.setColor(unselectedColor);
+
+                g.fillOval(verticesPoints[i].x - 3,
+                        verticesPoints[i].y - 3, 6, 6);
+            }
+        }
+        if (dragBoxRect != null) {
+            g.setColor(ink);
+            g.drawRect(dragBoxRect.x, dragBoxRect.y, dragBoxRect.width, dragBoxRect.height);
+        }
+        if (manipulator != null)
+            manipulator.paint(g);
+    }
+
+    private void drawGrid(Graphics2D g)
+    {
+        if (! parent.drawGrid())
+            return;
+        if (scale < 2.0)
+            return;
+
+        // Must use the original AT from the Graphics2D because at startup the coordinates
+        // are measured from the content pane of the UVMappingEditorDialog. Later on a new AT would do.
+
+        AffineTransform aBefore = g.getTransform();
+        AffineTransform aNow = new AffineTransform(aBefore);
+        Point corner = VertexToLayout(new Vec2(0, 0));
+        aNow.translate(corner.x, corner.y);
+        aNow.scale(scale, -scale);
+        g.setTransform(aNow);
+        g.setStroke(new BasicStroke((float)(1.0/scale)));
+
+        //double opacy = Math.min(255*Math.sqrt(scale/640), 223);
+        double opacy = Math.min(255.0*Math.pow(scale/520.0, 2.0/3.0), 223.0);
```

Comment:
Deliberate abbreviation? Or did you mean to write 'opacity'?

Reply:
> Or did you mean to write 'opacity'?

Should probaply be 'opacity' ... or just  'opac'.